pandas/core/base.py

```diff
@@ -1225,7 +1227,12 @@ def factorize(self, sort: bool = False, na_sentinel: int | None = -1):
         """
 
     @doc(_shared_docs["searchsorted"], klass="Index")
-    def searchsorted(self, value, side="left", sorter=None) -> np.ndarray:
+    def searchsorted(
+        self,
+        value: NumpyValueArrayLike,
+        side: Literal["left", "right"] = "left",
+        sorter: NumpySorter = None,
+    ) -> np.ndarray:
```

Comment:
this can sometimes return an int?

Reply:
`searchsorted` returns an `np.ndarray` according to the numpy docs and the signature in numpy 1.20.

Reply:
looks like that doc/signature is inaccurate then

```
>>> arr = np.arange(5)
>>> type(arr.searchsorted(3))
<class 'numpy.int64'>
```

Reply:
So this seems to be an issue with `numpy`.   This is using numpy 1.20.2:
```python
import numpy as np
a = np.array([1, 3, 5])
# reveal_type(a)
s = a.searchsorted(4)
# reveal_type(s)
print(s, type(s), isinstance(s, np.ndarray))
```

Result of running this is:
```
2 <class 'numpy.int64'> False
```
If I uncomment the `reveal_type` lines, the types of `a` and `s` are both `np.ndarray`.

So that gives us this question:

Should the pandas `searchsorted` API match what `numpy` documents (only array elements allowed), or what `numpy` implements?

I've submitted a numpy issue here:  https://github.com/numpy/numpy/issues/19160